### PR TITLE
Remove CSV parsing code

### PR DESCRIPTION
**Pull request by medley56**
# Remove CSV Definition

Removed the CSV Definition class entirely and all associated deprecated methods on the parser. Converted the parsing test for CTIM to use an XTCE definition instead of a CSV definition.

## Checklist
- [x] Changes are fully implemented without dangling issues or TODO items
- [x] Deprecated/superseded code is removed or marked with deprecation warning
- [NA] Current dependencies have been properly specified and old dependencies removed
- [x] New code/functionality has accompanying tests and any old tests have been updated to match any new assumptions
- [x] The changelog.md has been updated

### CODE REVIEW

**Review by greglucas — APPROVED** [review]
Looks good to me, some minor nit comments you can take/leave as you like.

File: space_packet_parser/parameters.py [hunk 1]
```diff
@@ -25,8 +25,12 @@ def __init__(self, name: str, encoding: encodings.DataEncoding, unit: Optional[s
         unit : Optional[str]
             String describing the unit for the stored value.
         """
+        if name is None:
+            raise ValueError("Parameter Type name attribute is required.")
         self.name = name
         self.unit = unit
```
[the remark this answers is not in the document]
Comment (medley56) [reply space_packet_parser/parameters.py:31]: It wasn't happening with name but it was happening with encoding. If the XML parser doesn't find an encoding element, it just returns None, which was causing extremely cryptic errors.

File: tests/test_data/ctim/ctim_xtce_v1.xml
[the remark this answers is not in the document]
Comment (medley56) [reply tests/test_data/ctim/ctim_xtce_v1.xml:1]: Yeah, it contains definitions for 38 APIDs. I could remove some of those and reduce it significantly but I wanted Mike to be able to use it.

File: space_packet_parser/parameters.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (medley56) [reply space_packet_parser/parameters.py:31]: Also, this only happens once per Parameter. It's not in the parsing loop so I think the performance impact should be negligible.

**Pull request merged**